spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -188,6 +187,7 @@ public Filter[] pushedFilters() {
   }
 
   @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
You have removed a decent amount of the branch points that were added in the prior commit that made this `SuppressWarnings` necessary. Can you see if its still needed after the most recent refactoring?

Reply:
Removed. Thanks!